src/main/java/org/antlr/jetbrains/adapter/parser/AntlrParseTreeToPsiConverter.java

```diff
@@ -183,7 +184,7 @@ public void exitEveryRule(ParserRuleContext ctx) {
                 marker.error("syntax error");
             }
         } else {
-            marker.done(getRuleElementTypes().get(ctx.getRuleIndex()));
+            marker.done((IElementType) getRuleElementTypes().get(ctx.getRuleIndex()));
```

Comment:
If only Java had intersection types, it would be possible to make this statically safe (i.e. the real type is
`? extends RuleIElementType & IElementType`, but alas! we have to get along with the assertion in PsiElementTypeFactory.